optuna/logging.py

```diff
@@ -28,15 +30,16 @@
 _default_handler: Optional[logging.Handler] = None
 
 
-def create_default_formatter() -> colorlog.ColoredFormatter:
+def create_default_formatter() -> Union[Formatter, colorlog.ColoredFormatter]:
```

Comment:
`colorlog.ColoredFormatter` extends `logging.Formatter`, and I guess the return type can be `Formatter`.

```suggestion
def create_default_formatter() -> Formatter:
```

And, the returned value is pass to `logging.Handler.setFormatter`, and it expects `logging.Formatter`. So, I don't think it does not depend on the `colorlog` version.

https://github.com/optuna/optuna/blob/ea55b34f9205d3e52fd4d825d4bf89c9daaf6696/optuna/logging.py#L70